src/formatDate.js

```diff
@@ -8,7 +8,46 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const arrayFromDate = date.split(fromFormat[fromFormat.length - 1]);
+  const arrForResulf = [];
+  const objectFromDate = {};
+
+  for (let i = 0; i < arrayFromDate.length; i++) {
+    objectFromDate[fromFormat[i]] = arrayFromDate[i];
+  }
+
+  const { MM: month, DD: day, ...lastElement } = objectFromDate;
+  const [year] = Object.values(lastElement);
+
+  for (const y of toFormat) {
+    if (y === 'YYYY' || y === 'YY') {
+      if (year.length === y.length) {
+        arrForResulf.push(year);
+      }
+
+      if (year.length > y.length) {
+        arrForResulf.push(year.slice(-2));
+      }
+
+      if (year.length < y.length) {
+        if (year < 30) {
+          arrForResulf.push('20' + year);
+        } else {
+          arrForResulf.push('19' + year);
```

Comment:
The logic for handling years less than 30 assumes that they belong to the 21st century, while others belong to the 20th century. This assumption might not hold for all cases. Consider providing a more flexible way to handle century determination.